src/tests/adding.md

```diff
@@ -29,11 +29,12 @@ rough heuristics:
   - need to run gdb or lldb? use the `debuginfo` test suite
   - need to inspect LLVM IR or MIR IR? use the `codegen` or `mir-opt` test
     suites
-  - need to run rustdoc? Prefer a `rustdoc` test
+  - need to run rustdoc? Prefer a `rustdoc` or `rustdoc-ui` test. Occasionally you'll need `rustdoc-js` as well.
   - need to inspect the resulting binary in some way? Then use `run-make`
+- Library tests should go in `library/$crate/tests` (where `$crate` is usually `core`, `alloc`, or `std`).
```

Comment:
I usually prefer to add `${brackets}` in a case like this, rather than hoping the shell parses it correctly. :^) Actually, it might even be better to strictly match Rust's `fmt` syntax, if we're going to use shorthand...

Would be good to mention that a "library test" is one that tests an API's functionality directly, merely verifying it succeeds or has some simple behavior... basically, to specify that `run-pass` tests should usually be library tests, without directly saying "don't add run-pass tests".